modular_skyrat/modules/morgue_radio/morgue.dm

```diff
@@ -0,0 +1,36 @@
+/obj/structure/bodycontainer/morgue/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/morgue_radio)
+
+/obj/structure/bodycontainer
+	var/obj/item/radio/radio // The internal radio in the morgue trays.
+
+/datum/component/morgue_radio
+	var/obj/structure/bodycontainer/morgue/morgue // Typecasted reference to the current tray.
+
+
+/datum/component/morgue_radio/RegisterWithParent()
+	morgue = parent
+	morgue.radio = new /obj/item/radio(morgue) // Initialize the radio in the morgue tray
```

Comment:
Machines usually use the appropriate headset, in this case `/obj/item/radio/headset/headset_med` and then they can `talk_into()` the appropriate channel and not do hacks with manually doing set_frequency()